scheduler/src/cook/tools.clj

```diff
@@ -907,14 +907,23 @@
     (cond-> {:count 1 :cpus cpus :mem mem}
       gpus (assoc :gpus gpus))))
 
+(defn match-based-on-regexp
+  "Given a list of dictionaries [{:<regexp-name> <regexp> :<field-name> <field>} {:<regexp-name> <regexp> :<field-name> <field>} ...], match-list,
+   a key <field-name> and <regexp-name> name, return the first matching <field> where the <regexp> matches the key."
+  [regexp-name field-name match-list key]
+  (->> match-list
+       (filter (fn [map]
+                 (let [regexp (get map regexp-name)
+                       pattern (re-pattern regexp)]
+                   (re-find pattern key))))
+       first
+       field-name))
```

Comment:
let's change to 
(-> match-list
       (->> (filter (fn [map]
                 (let [regexp (get map regexp-name)
                       pattern (re-pattern regexp)]
                   (re-find pattern key)))))
       first
       (get field-name))